posthog/temporal/batch_exports/spmc.py

```diff
@@ -256,20 +256,14 @@ async def start(
                             "Empty queue with no more events being produced, closing writer loop and flushing"
                         )
                         self.flush_start_event.set()
-                        # Exit context manager to trigger flush
+                        # Exit context manager to trigger final flush
                         break
                     else:
-                        await asyncio.sleep(0.1)
+                        await asyncio.sleep(0)
```

Comment:
I've noticed we do `await asyncio.sleep(0)` in a few places - is this basically to hand control back over to the event loop?

Reply:
Yeah, otherwise if we didn't we would loop forever here and never let the producer put stuff in the queue.

Reply:
Similar reasons in other places: We are waiting for other tasks to progress.

Reply:
Not the cleanest API for synchronization between tasks, we can later look at semaphores or barriers.

Reply:
nice, thanks